### DISCUSSION TITLE
Work with large files

### DISCUSSION THREAD

**Discussion by vHanda** [GitJournal/GitJournal#789] [Feature Requests]
GitJournal is currently incapable of working with large files. It slows down dramatically. Fixing this would require at-least two big changes -

1. Not load all the notes in memory. GitJournal should only load the required notes into memory. This would also improve performance in lower end devices.

2. Use a proper data-structure for a text editor, instead of one large string which gets modified each time. This would require us to not use Flutter's default text editor and implement our own.

**Comment by bsecker** [comment]
I'm very much affected by this.  Another thing I want to note - I tried this app with my large Foam workspace (1Gb+) and the entire user interface is sluggish with the amount of notes I have (780 ish), with a deep folder structure (5+ directories deep in places).

I don't know what kind of testing/performance reporting I am able to do on my phone - is there any stats/info you'd like me to collect? Should I create a new issue for this?
